tests/worker/test_workflow.py

```diff
@@ -0,0 +1,989 @@
+
+
+@workflow.defn
+class CancelActivityWorkflow:
+    @workflow.run
+    async def run(self, wait_for_cancel: bool) -> str:
+        cancellation_type = workflow.ActivityCancellationType.TRY_CANCEL
+        if wait_for_cancel:
+            cancellation_type = (
+                workflow.ActivityCancellationType.WAIT_CANCELLATION_COMPLETED
+            )
+        handle = workflow.start_activity(
+            wait_cancel,
+            schedule_to_close_timeout=timedelta(seconds=5),
+            heartbeat_timeout=timedelta(seconds=1),
+            cancellation_type=cancellation_type,
+        )
+        await asyncio.sleep(0.01)
+        handle.cancel()
+        return await handle
+
+
+async def test_workflow_cancel_activity(client: Client):
+    async with new_worker(
+        client, CancelActivityWorkflow, activities=[wait_cancel]
+    ) as worker:
+        # Do not wait for cancel
+        with pytest.raises(WorkflowFailureError) as err:
+            await client.execute_workflow(
+                CancelActivityWorkflow.run,
+                False,
+                id=f"workflow-{uuid.uuid4()}",
+                task_queue=worker.task_queue,
+            )
+        assert isinstance(err.value.cause, ActivityError)
+        assert isinstance(err.value.cause.cause, CancelledError)
+
+        # Wait for cancel
+        result = await client.execute_workflow(
+            CancelActivityWorkflow.run,
+            True,
+            id=f"workflow-{uuid.uuid4()}",
+            task_queue=worker.task_queue,
+        )
+        assert "Got cancelled error, cancelled? True" == result
+
+
+@activity.defn
+async def wait_local_cancel() -> str:
+    try:
+        await asyncio.sleep(1000)
+        raise RuntimeError("Should not get here")
+    except asyncio.CancelledError:
+        return "Got cancelled error, cancelled? " + str(activity.is_cancelled())
+
+
+@workflow.defn
+class CancelLocalActivityWorkflow:
+    @workflow.run
+    async def run(self, wait_for_cancel: bool) -> str:
+        cancellation_type = workflow.ActivityCancellationType.TRY_CANCEL
+        if wait_for_cancel:
+            cancellation_type = (
+                workflow.ActivityCancellationType.WAIT_CANCELLATION_COMPLETED
+            )
+        handle = workflow.start_local_activity(
+            wait_local_cancel,
+            schedule_to_close_timeout=timedelta(seconds=5),
+            cancellation_type=cancellation_type,
+        )
+        await asyncio.sleep(0.01)
+        handle.cancel()
+        return await handle
+
+
+# # @pytest.mark.skip(reason="Make test to timeout WFT better")
+async def test_workflow_cancel_local_activity(client: Client):
+    async with new_worker(
+        client, CancelLocalActivityWorkflow, activities=[wait_local_cancel]
+    ) as worker:
+        # Do not wait for cancel
+        with pytest.raises(WorkflowFailureError) as err:
+            await client.execute_workflow(
+                CancelLocalActivityWorkflow.run,
+                False,
```

Comment:
nit: use keyword argument so it's clear what is `False`

Reply:
We don't support keyword arguments in SDKs. Were this something besides a test case, I'd make params a dataclass.